tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/device/DeviceSpec.java

```diff
@@ -0,0 +1,162 @@
+package org.tensorflow.device;
+
+import javax.annotation.Nullable;
```

Comment:
Where do we get the Nullable annotation from?

Reply:
I was wondering as well, I assumed (maybe a bit too fast) that it was part of the JDK... ok, I'll remove them manually and merge back the file without it

Reply:
Yeah, it was coming again from our dependency to the `tensorflow-core-generator`, which got it from `com.google.code.findbugs`, thanks for raising this @Craigacp

Reply:
Woah I always thought @Nullable was part of the JDK somewhere...
My bad for taking that for granted.
Are you folks already removing that?

Reply:
No worries. Can we remove the generator dependency somehow? It pulls in a lot of things.

Reply:
Personally I'm a bit confused as to why jsr305 wasn't integrated into the JDK, but there was a talk about nullness annotations from Google at the last JVMLS discussing all the issues with integrating them into things, and it seemed to be pretty complex. Unfortunately that talk wasn't recorded.

Reply:
> Can we remove the generator dependency somehow?

I was thinking of doing that as well but there is probably a good reason why it is there, probably maven was failing somehow to run the annotation processor without it. Will need to double check

So @hthu , it's all good now, I forced-pushed a new version with `@Nullable`. I also took the opportunity to move the `DeviceSpec` out of the `device` package. Since there was only one file in that package, I don't think we needed it.

Reply:
Sounds good to me!
Thank you both!

Reply:
> > Can we remove the generator dependency somehow?
> 
> I was thinking of doing that as well but there is probably a good reason why it is there, probably maven was failing somehow to run the annotation processor without it. Will need to double check

I guess without the explicit dependency it's up to maven which one of the generator and the core api it compiles first.  Maybe we could make it a test time dependency? That's even less clear in terms of the semantics, but it should make it run first.